### ISSUE TITLE
[Feature Request] Player auto-rotate

### ISSUE THREAD

**Issue by black-cerberon** [TeamNewPipe/NewPipe#4140]
Please add forced auto-rotate in video player.

**Comment by test2a** [comment]
I would love to have this apply for vertical videos. Changing the orientation in Newpipe makes it default and watching a regular video later I have to turn it back on.

**Comment by opusforlife2** [comment]
Try this https://github.com/TeamNewPipe/NewPipe/files/5079507/app-debug.zip and reply if you still need forced autorotate.

**Comment by black-cerberon** [comment]
@opusforlife2 that's work but better rotate fullscreen media by device rotation.

**Comment by opusforlife2** [comment]
Can you rephrase that? I didn't get it.

**Comment by black-cerberon** [comment]
Fullscreen media can use system rotation setting, sensor state or bind only one orientation like you implemented right now. Usally many apps read system rotation setting for change screen orientation. When use sensor state app can ignore system setting and rotate everytime if device was rotated. For example you can look at Simple Gallery App - https://github.com/SimpleMobileTools/Simple-Gallery

**Comment by opusforlife2** [comment]
I'm asking why do you _need_ sensor based rotation? In the apk I linked above, when you tap on the fullscreen button, if you're watching a vertical video, the video stays in portrait mode, and if you're watching a normal video, then the video rotates to landscape mode.

Where is the need to use sensor state?

**Comment by black-cerberon** [comment]
Sorry, I not tested it with vertical videos.

**Comment by opusforlife2** [comment]
It's alright. I was just looking for clarification because I suspected your issue might be already solved.

There could _still_ be some users who want sensor based rotation separate from system rotation. If that is the case, it would be up to the devs whether or not to implement it. Or maybe a contributor could decide to take it up.

**Comment by test2a** [comment]
> Try this https://github.com/TeamNewPipe/NewPipe/files/5079507/app-debug.zip and reply if you still need forced autorotate.

@opusforlife2 this version did the trick although it seems the rotate button is no longer there, not that we need it. What does the globe button do ? i clicked on it and the app crashed for me
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Country:__ IN
* __Content Language:__ en-IN
* __App Language:__ en_IN
* __Service:__ none
* __Version:__ 0.19.8
* __OS:__ Linux Android 7.0 - 24
<details><summary><b>Crash log </b></summary><p>

```
android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=https://www.youtube.com/... flg=0x10000000 pkg=com.zui.resolver }
	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1816)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1525)
	at android.app.Activity.startActivityForResult(Activity.java:4265)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:675)
	at android.app.Activity.startActivityForResult(Activity.java:4224)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:662)
	at android.app.Activity.startActivity(Activity.java:4548)
	at android.app.Activity.startActivity(Activity.java:4516)
	at org.schabi.newpipe.util.ShareUtils.openUrlInBrowser(ShareUtils.java:37)
	at org.schabi.newpipe.player.VideoPlayerImpl.onOpenInBrowserClicked(VideoPlayerImpl.java:910)
	at org.schabi.newpipe.player.VideoPlayerImpl.onClick(VideoPlayerImpl.java:790)
	at android.view.View.performClick(View.java:5617)
	at android.view.View$PerformClick.run(View.java:22271)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6153)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:868)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:758)

```
</details>
<hr>

**Comment by opusforlife2** [comment]
@test2a If you lock device rotation, the full screen button will appear.

The globe button is to Open In Browser. Your error is the same as in #3925. Can you supply the information that Stypox asked for there?

**Comment by test2a** [comment]
@opusforlife2 yes. I have not set a default browser on the phone. When clicking on a link anywhere else, it asks to choose the browser to open this link in. 
Now that i have set the default browser the error is not appearing. I did write on the link about having a proper error message instead of a crash, lets see
